### ISSUE TITLE
Simplify object navigation

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3931]
**Reported by k_kolev1985 on 2014-02-26 12:21**
I've been playing around with Narrator under Windows 8.1 recently and I've discovered that its object navigation is in some ways much simpler than the one presently used in NVDA. The reason: it does not have parent and children levels - all objects are on one single level. The advantage of this is that the user does not have to move through several levels of objects, just to find an object, "hidden" in some sub-sub-sub-children level - he will have to only move through the objects one by one, until he finds the one he is looking for. The disadvantage is that sometimes the number of objects, if placed on one single level, may become in large numbers and moving through them one by one may take some time. Even so however, if the user is not familiar with the hierarchy of the objects in an application (and believe me - most users are not), the possibility to move through all objects in one single level will be easier for them. Can it be made so NVDA provides such a mode of navigation through the objects - all on one level (at least in an application's window)? Of course, this mode should be optional and toggled for example from the "Review cursor settings" window with another option added there.

**Comment by bhavyashah** [comment]
@josephsl You seemed to suggest in https://github.com/nvaccess/nvda/issues/3931#issuecomment-155317794 that you may consider working on this ticket in the future. This is mostly just a friendly reminder to hopefully get around to working on this as and when time, circumstance and priorities permit.

**Comment by ehollig** [comment]
Possibly related: #4100

**Comment by leonardder** [comment]
Coming back to this, I think that it would be beneficial to have keyboard bindings for flattened object navigation indeed. Since shift+NVDA+pageUp and shift+NVDA+pageDown are still unassigned, I think it might make sense to use these for the laptop layout. For the Desktop layout, I'd like to suggest NVDA+numpad3 and NVDA+numpad9 for next and previous object, respectively.

**Comment by Adriani90** [comment]
Also this could be covered by locking object Navigation and assigning quick Navigation key like in browse mode.

**Comment by Adriani90** [comment]
see #7759

**Comment by josephsl** [comment]
Hi,

There are several solutions:

1. Assign commands suggested by @LeonardDer: NVDA+page up/page down could work, although to me, it doesn't make sense. NVDA+arrows (NVDA+Shift+arrows on laptop layout) are already assigned to object navigation, and for most users, it is part of their muscle memory. I was going to suggest NVDA+Shift+Numpad left/right arrows for flow command for desktop layout, but then it doesn't work well for laptop layout (it also runs against personal philosophy of user experience consistency).
2. Add a setting in Review Cursor panel to control how next and previous object navigation should work. Default would be current behavior, and users can choose to enable flow method. The biggest benefit is that one can toggle this setting on per application basis (via configuration profiles), but the downside is redundant touchscreen gestures. One way to solve touch gesture problem is detecting gesture type (isinstance) and keep the original behavior if this is touch only. Also, in order to make this more effective, an unassigned toggle command should be added.

Thanks.

**Comment by derekriemer** [comment]
[**@josephsl**](https://github.com/josephsl) commented on [Jun 25, 2018, 12:46 AM MDT](https://github.com/nvaccess/nvda/issues/3931#issuecomment-399848917 "2018-06-25T06:46:21Z - Replied by Github Reply Comments"):
> Hi,
> 
> There are several solutions:
> 
> 1. Assign commands suggested by @LeonardDer: NVDA+page up/page down could work, although to me, it doesn't make sense. NVDA+arrows (NVDA+Shift+arrows on laptop layout) are already assigned to object navigation, and for most users, it is part of their muscle memory.

Having trained various NVDA users in the past object navigation is not even close to intuitive to most users, especially unintuitive to non computer scientists. Most users sinply don't use object navigation out of frustration with using Object navigation. They'll listen to a whole dialog go by with NVDA+b for a good 30 seconds because object nav is so unintuitive. Something needs done about this.

> As I suggested three years ago, NVDA+Shift+Numpad left/right arrows for flow command for desktop layout was what I was going to suggest, but then it doesn't work well for laptop layout (it also runs against personal philosophy of user experience consistency).

I respectfully prefer @leonardder approach because it's a simpler keystroke.

> 2. Add a setting in Review Cursor panel to control how next and previous object navigation should work. Default would be current behavior, and users can choose to enable flow method. The biggest benefit is that one can toggle this setting on per application basis (via configuration profiles), but the downside is redundant touchscreen gestures. One way to solve touch gesture problem is detecting gesture type (isinstance) and keep the original behavior if this is touch only. Also, in order to make this more effective, an unassigned toggle command should be added.

I agree with this proposal, but per my comments above (with anecdotal evidence however) I think an flows to mode should be the default.

>

**Comment by CyrilleB79** [comment]
Initial request from @k-kolev1985 fixed in #15065.
